test/extensions/filters/network/common/redis/client_impl_test.cc

```diff
@@ -99,6 +105,134 @@ class RedisClientImplTest : public testing::Test, public Common::Redis::DecoderF
   ClientPtr client_;
 };
 
+TEST_F(RedisClientImplTest, BatchWithZeroBufferAndTimeout) {
+  // Basic test with a single request, default buffer size (0) and timeout (0).
+  // This means we do not batch requests, and thus the flush timer is never enabled.
+  InSequence s;
+
+  setup();
+
+  // Make the dummy request
+  Common::Redis::RespValue request1;
+  MockPoolCallbacks callbacks1;
+  EXPECT_CALL(*encoder_, encode(Ref(request1), _));
+  EXPECT_CALL(*flush_timer_, enabled()).WillOnce(Return(false));
+  PoolRequest* handle1 = client_->makeRequest(request1, callbacks1);
+  EXPECT_NE(nullptr, handle1);
+
+  // Process the dummy request
+  Buffer::OwnedImpl fake_data;
+  EXPECT_CALL(*decoder_, decode(Ref(fake_data))).WillOnce(Invoke([&](Buffer::Instance&) -> void {
+    InSequence s;
+    Common::Redis::RespValuePtr response1(new Common::Redis::RespValue());
+    EXPECT_CALL(callbacks1, onResponse_(Ref(response1)));
+    EXPECT_CALL(*connect_or_op_timer_, disableTimer());
+    EXPECT_CALL(host_->outlier_detector_, putResult(Upstream::Outlier::Result::SUCCESS));
+    callbacks_->onRespValue(std::move(response1));
+  }));
+  upstream_read_filter_->onData(fake_data, false);
+
+  EXPECT_CALL(*connect_or_op_timer_, disableTimer());
+  client_->close();
+}
+
+class ConfigBufferSizeGTSingleRequest : public Config {
+  bool disableOutlierEvents() const override { return false; }
+  std::chrono::milliseconds opTimeout() const override { return std::chrono::milliseconds(25); }
+  bool enableHashtagging() const override { return false; }
+  unsigned int maxBufferSizeBeforeFlush() const override { return 8; }
+  std::chrono::milliseconds bufferFlushTimeoutInMs() const override {
+    return std::chrono::milliseconds(1);
+  }
+};
+
+TEST_F(RedisClientImplTest, BatchWithTimerFiring) {
+  // With a flush buffer > single request length, the flush timer comes into play.
+  // In this test, we make a single request that doesn't fill the buffer, so we
+  // have to wait for the flush timer to fire.
+  InSequence s;
+
+  setup(std::make_unique<ConfigBufferSizeGTSingleRequest>());
+
+  // Make the dummy request
+  Common::Redis::RespValue request1;
+  MockPoolCallbacks callbacks1;
+  EXPECT_CALL(*encoder_, encode(Ref(request1), _));
+  EXPECT_CALL(*flush_timer_, enableTimer(_));
+  PoolRequest* handle1 = client_->makeRequest(request1, callbacks1);
+  EXPECT_NE(nullptr, handle1);
+
+  // Pretend the the flush timer fires
+  EXPECT_CALL(*flush_timer_, enabled()).WillOnce(Return(false));
+  ; // timer is disabled, as it has already fired
```

Comment:
@mattklein123 this was the python formatter. Should I flag this as an issue?

Reply:
What do you mean?